packages/babel-plugin-transform-parameters/test/fixtures/regression/scope-gen-async/output.js

```diff
@@ -0,0 +1,23 @@
+function f() {
+  let a = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : 1;
+  return function* (a) {
+    var a = yield a;
+    return a;
+  }(a);
+}
+
+function g() {
+  let a = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : 1;
```

Comment:
The default expression could actually throw a sync error, which needs to be caught sand turned into a rejected promise.